qemu/softmmu/memory.c

```diff
@@ -98,6 +98,9 @@ MemoryRegion *memory_cow(struct uc_struct *uc, MemoryRegion *current, hwaddr beg
     hwaddr current_offset;
     MemoryRegion *ram = g_new(MemoryRegion, 1);
 
+    assert((begin & ~TARGET_PAGE_MASK) == 0);
```

Comment:
These checks look not necessary considering you have aligned them before calling the function?

Reply:
These debug checks are to avoid adding the bug again. In release mode the they are noops. But when some change the code and miss the alignment, these checks can help to find the bug.